core/store/src/archive/cold_storage.rs

```diff
@@ -165,13 +166,15 @@ fn rc_aware_set(
 fn copy_state_from_store(
     shard_layout: &ShardLayout,
     block_hash_key: &[u8],
-    cold_db: &ColdDB,
```

Comment:
in my understanding `&Arc<T>` should not be used as an argument type:
* if you want to clone Arc internally then just use owned `Arc<T>`
* otherwise use &T as is being used here
Could you please explain you motivation for this change